src/search/mh_prog.cc

```diff
@@ -394,7 +401,14 @@ void mh_sampler::mcmc_iter(top_k_progs& topk_progs, int niter, prog* orig, bool
   prog_start = new prog(*orig);
   // set the error cost and perf cost of the prog start
   prog_start->_error_cost = 0;
-  _cost.perf_cost(prog_start, inst::max_prog_len);
+  if(k2_config.functionality == FUNC_optimize){
+    _cost.perf_cost(prog_start, inst::max_prog_len);
+  }else if (k2_config.functionality == FUNC_repair){
+    // _cost.safety_cost_repair(prog_start, inst::max_prog_len, prog_start, inst::max_prog_len);
+    // cout << "safety_cost_repair " << prog_start->_safety_cost << endl;
+    prog_start->set_safety_cost(100);
```

Comment:
What is the significance of 100, could you explain?
If there are k unsafe instructions in the original program, maybe just use `k` as the cost?
I recall having some discussion of the safety cost of the original program, but can't recall what we concluded now. Adding a comment describing the reasoning for the fixed cost may help. 

In general, it would be good to avoid magic numbers in the code. If you have to set a parameter to a fixed value, add a `#define` along with where other parameters are defined.

Reply:
The ideal goal is here is to get the complete safety cost of original program by executing:
_cost.safety_cost_repair(prog_start, inst::max_prog_len, prog_start, inst::max_prog_len);

However, this was giving some debugging issues. So, I think Qiongwen is looking at how to fix this. I'll check in if there is anyway I can help for this.

At the time just to get the code running, as a placeholder, we randomly initialized the safety cost of original program as 100, since it is a big number. No other real significance for 100.

Yeah, using the number of unsafe instructions seems like a better placeholder. Thanks! I changed the code to do this for now